com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -37,19 +37,41 @@ class _ResourceElement(object):
     """
     def __init__(self, json_rep, rest_client):
         """
-        :param json_rep: The JSON response from a REST call.
-        :param rest_client: The client used to make the REST call.
+        Args:
+            json_rep(dict): The JSON response from a REST call.
+            rest_client(_StreamsRestClient): The client used to make the REST call.
         """
-        self.rest_client=rest_client
-        for key in json_rep:
-            if key == 'self':
-                self.__dict__["rest_self"] = json_rep['self']
-            else:
-                self.__dict__[key] = json_rep[key]
+        self.rest_client = rest_client
+        self.rest_self = json_rep.get('self', None)
+        self.json_rep = json_rep
 
     def __str__(self):
         return pformat(self.__dict__)
 
+    # Override getattr to retrieve attribute from response JSON
+    def __getattr__(self, key):
+        if 'json_rep' in self.__dict__:
```

Comment:
I think `__getattr__` is only called if the attribute is not found in the object.   So `self.__dict__` will call `__getattribute__`, but not `__getattr__`